plugins/module_utils/network/ios/argspec/l3_interfaces/l3_interfaces.py

```diff
@@ -92,6 +94,7 @@ class L3_interfacesArgs(object):  # pylint: disable=R0903
             },
         },
         "running_config": {"type": "str"},
+        "restore_commands": {"type": "bool", "default": "False"},
```

Comment:
@bentole I know this PR is WIP, but restore_commands would be not accepted as a part of the documentation as Ansible argspec doesn't have any declaration for restore_commands, and adding an ad-hoc attribute exclusively to a module is strictly not recommended! The top-level attributes are to be maintained consistently for sake of an argspec being valid for Ansible to recognize it. I am unsure about the functionality you are trying to achieve. Would share some more details regarding it.
You might see sanity already complaining about it.
Regards

Reply:
Hi @KB-perByte ! The CI is happy now. The restore_commands is explained in the restoration paragraph. I do understand the concern about breaking the consistency of top-level attributes. I need some help on this. I first had the option together with the vrf parameter so instead of vrf being a string I had

```
 "vrf": {
        "type": "dict",
        "options": {
            "name": {"type": "str"},
            "restore_commands": {"type": "bool"},
        },
}
```
But it felt wrong because restore_commands is referenced to any router command or config. Or is that ok? On top of it all I feel the restoration shouldnt be in the offical release at all due the complexity it adds to the module. On the other hand I believe it's rather cumbersome, if not impossible, to perform consistent restoration using separate tasks (?) At least to accomodate for all eventualities. Being a bit of a devils advocate here 😄

Reply:
@bentole this is interesting, but what I can figure out is restore_command would never fire a specific configuration nor flow into facts when we retrieve it. Moreover, the usecase of it is specific to how the config must be applied in when vrf is present. I was just wondering how can we not have a set of logic in the code if we need to do it. And not have a user-facing attribute that changes the course of the whole module execution when enabled.
Note - Facts that are generated are considered source of truth and an attribute not getting enabled/ disabled when facts are generated would have a good impact on the round tripping functionality that the resource module support.
We shall discuss more on this.
Regards.